mne_bids/write.py

```diff
@@ -1001,157 +998,155 @@ def write_raw_bids(raw, bids_path, events_data=None,
+                            bids_path.root, raw, bids_path.datatype,
+                            overwrite, verbose)
     else:
-        logger.warning(f'Writing of electrodes.tsv '
-                       f'is not supported for datatype "{datatype}". '
-                       f'Skipping ...')
+        logger.warning(f'Writing of electrodes.tsv is not supported '
+                       f'for datatype "{bids_path.datatype}". Skipping ...')
 
     events, event_id = _read_events(events_data, event_id, raw, ext,
                                     verbose=verbose)
     if events is not None and len(events) > 0 and not emptyroom:
-        _events_tsv(events, raw, events_fname, event_id, overwrite, verbose)
+        _events_tsv(events, raw, events_path.fpath, event_id,
+                    overwrite, verbose)
 
-    make_dataset_description(bids_root, name=" ", overwrite=overwrite,
+    make_dataset_description(bids_path.root, name=" ", overwrite=overwrite,
                              verbose=verbose)
 
-    _sidecar_json(raw, task, manufacturer, sidecar_fname, datatype, overwrite,
-                  verbose)
-    _channels_tsv(raw, channels_fname, overwrite, verbose)
+    _sidecar_json(raw, bids_path.task, manufacturer, sidecar_path.fpath,
+                  bids_path.datatype, overwrite, verbose)
+    _channels_tsv(raw, channels_path.fpath, overwrite, verbose)
 
     # create parent directories if needed
     _mkdir_p(os.path.dirname(data_path))
 
-    if os.path.exists(bids_fname) and not overwrite:
-        raise FileExistsError(f'"{bids_fname}" already exists. '  # noqa: F821
+    if os.path.exists(bids_path.fpath) and not overwrite:
+        raise FileExistsError(f'"{bids_path.fpath}" already exists. '  # noqa: F821
                               f'Please set overwrite to True.')
 
     # If not already converting for anonymization, we may still need to do it
     # if current format not BIDS compliant
     if not convert:
-        convert = ext not in ALLOWED_DATATYPE_EXTENSIONS[datatype]
+        convert = ext not in ALLOWED_DATATYPE_EXTENSIONS[bids_path.datatype]
 
-    if datatype == 'meg' and convert and not anonymize:
+    if bids_path.datatype == 'meg' and convert and not anonymize:
         raise ValueError(f"Got file extension {convert} for MEG data, "
                          f"expected one of "
                          f"{ALLOWED_DATATYPE_EXTENSIONS['meg']}")
 
     if not convert and verbose:
-        print('Copying data files to %s' % op.splitext(bids_fname)[0])
+        print('Copying data files to %s' % op.splitext(bids_path.fpath)[0])
 
     # File saving branching logic
     if convert:
-        if datatype == 'meg':
-            if ext == '.pdf':
-                bids_fname = op.join(data_path, op.basename(bids_fname))
-            _write_raw_fif(raw, bids_fname)
+        if bids_path.datatype == 'meg':
+            _write_raw_fif(
+                raw, (op.join(data_path, op.basename(bids_path.fpath))
```

Comment:
we should ideally adopt Path way of doing things and not rely on os.path 

maybe for later.

```suggestion
                raw, data_path / bids_path.fpath.name
```